### DISCUSSION TITLE
Modbusbridge and additional requests to the ClientRTU

### DISCUSSION THREAD

**Discussion by JustMe-NL** [eModbus/eModbus#288] [Q&A] [answered]
If got a esp32s3 working as a ModbusWiFiBridge to two Atmega32u4 servers over RS485 (using SimpleModbus NG on the Atmega's by Angeloc). Apart from bridging the ESP32 also has several registered workers to do some work on it's own, so basically for the outside world we have 3 servers reachable by Modbus TCP.

So far so good but next I need my programming on the ESP32 to also be able to communicate with the servers without outside orders. As far as I can tell I can't just add a onData and OnError handler to the existing ModbusClientRTU because they are in use by the bridge. However, am I right in assuming I could send and recieve my own messages to the RTU servers by utilizing the syncRequest function or does that still interfere with the working of the standard Bridge?

**Comment by Miq1** [comment, marked as answer]
You are right, ``syncRequest()`` may have issues because there are remote requests being processed by the bridge that are using a specialized ``syncRequest()`` themselves.

But you should try ``ModbusMessage response = bridge.localRequest(ModbusMessage request);`` instead. Despite the name 'local', if the addressed server is one being bridged, it will be called as if the request would have been received from a remote client. The call is blocking like ``syncRequest()``, since it needs to directly return the response.

**Reply by JustMe-NL** [reply]
I will try that, thank you for the extreem quick response.

**Reply by JustMe-NL** [reply]
Tried it and it works perfectly for what I needed. Thank you for all your hard work you put into this.

**Comment by Miq1** [comment]
🏆
